src/sentry/static/sentry/app/components/scoreBar.tsx

```diff
@@ -4,7 +4,21 @@ import styled from '@emotion/styled';
 
 import theme from 'app/utils/theme';
 
-class ScoreBar extends React.Component {
+type DefaultProps = {
+  size: number;
+  thickness: number;
+  radius: number;
+  palette: Readonly<string[]>;
+};
+
+type Props = DefaultProps & {
+  vertical?: boolean;
+  score: number;
+  paletteClassNames?: string[];
+  className?: string;
+};
+
+class ScoreBar extends React.Component<Props> {
```

Comment:
Since you have no state in this component converting it to a function component could help make the typescript conversion simpler as you could use default values in the destructure instead of defaultProps.

Reply:
Oh! Good call, thanks.